lib/components/app/app-frame.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { Col, Row } from 'react-bootstrap'
+
+import DesktopNav from './desktop-nav'
+
+/**
+ * This component defines the general application frame, to which
+ * content and an optional sub-navigation component can be inserted.
+ */
+const AppFrame = ({ children, SubNav }) => (
+  <div className='otp'>
+    {/* TODO: Do mobile view. */}
+    <DesktopNav />
+    {SubNav && <SubNav />}
+    <div className='container'>
+      <Row xs={12}>
+        <Col md={8} mdOffset={2} sm={10} smOffset={1} xs={12}>
+          {children}
+        </Col>
+      </Row>
+    </div>
+  </div>
+)
+
+/**
+ * Creates a simple wrapper component consisting of an AppFrame that surrounds
+ * the provided component. (Displays "no content" if none provided.)
+ */
+export function frame (Component) {
+  return () => (
+    <AppFrame>
+      {Component
+        ? <Component />
+        : <p>No content provided.</p>
```

Comment:
Maybe we should create a `PageNotFound` component and render that if Component does not exist (it could have a link back to the main trip planner).

Reply:
Hmm I have set these pages to open in new tabs, so does a link make sense with this setup?

Reply:
Yea, perhaps no link in that case.

Reply:
I enhanced the format for not found content in https://github.com/opentripplanner/otp-react-redux/pull/340/commits/f95638817c7736ad80cb0d6a83f1e52ee3581ebc.